openmdao/vectors/vector.py

```diff
@@ -292,6 +292,17 @@ def _contains_abs(self, abs_name):
         """
         return abs_name in self._names
 
+    def keys(self):
+        """
+        Return absolute variable names of variables contained in this vector.
+
+        Returns
+        -------
+        list
+            absolute variable names of variables contained in this vector.
+        """
+        return self._views.keys()
```

Comment:
keys() should use self._names instead of self._views because the active set of what is contained in the vector changes depending on whether the vector is inside of a matvec_context or not, and that active set is specified by self._names.